### PR TITLE
apps/example/esp_wifi_demo: add esp32 wifi demo.

### PR DESCRIPTION
**Pull request by ChinaTizenRT**
This esp32 wifi demo call wifi driver interface directly.

### CODE REVIEW

File: apps/examples/esp32_wifi_demo/station.c [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+#define DEFAULT_AUTHMODE WIFI_AUTH_WEP
+#elif CONFIG_EXAMPLE_WPA
+#define DEFAULT_AUTHMODE WIFI_AUTH_WPA_PSK
+#elif CONFIG_EXAMPLE_WPA2
+#define DEFAULT_AUTHMODE WIFI_AUTH_WPA2_PSK
+#else
+#define DEFAULT_AUTHMODE WIFI_AUTH_OPEN
+#endif
+#else
+#define DEFAULT_RSSI -127
```
Comment (SRCN-LiQin) [inline apps/examples/esp32_wifi_demo/station.c]: better add () for negative value in macro

Comment (ChinaTizenRT) [reply apps/examples/esp32_wifi_demo/station.c]: right, done.

File: apps/examples/esp32_wifi_demo/Makefile [hunk 2]
```diff
@@ -0,0 +1,122 @@
+###########################################################################
+#
+# Copyright 2018 Samsung Electronics All Rights Reserved.
```
Comment (xixidodo) [inline apps/examples/esp32_wifi_demo/Makefile:3]: year 2019

File: apps/examples/esp32_wifi_demo/Makefile [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+###########################################################################
+-include $(TOPDIR)/.config
+-include $(TOPDIR)/Make.defs
+include $(APPDIR)/Make.defs
+
+# Hello, World! built-in application info
```
Comment (xixidodo) [inline apps/examples/esp32_wifi_demo/Makefile:22]: WiFi demo

File: apps/examples/esp32_wifi_demo/Makefile [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+PRIORITY = SCHED_PRIORITY_DEFAULT
+STACKSIZE = 2048
+
+# Hello, World! Example
```
Comment (xixidodo) [inline apps/examples/esp32_wifi_demo/Makefile:27]: same as above

**Pull request merged**